encoder.go

```diff
@@ -36,7 +36,16 @@ func newXMLEncoder(w io.Writer, floatPresicion int) *xmlEncoder {
 // AddRelationship adds a relationship to the encoded model.
 // Duplicated relationships will be removed before encoding.
 func (enc *xmlEncoder) AddRelationship(r spec.Relationship) {
-	enc.relationships = append(enc.relationships, Relationship(r))
+	hasRelationship := false
```

Comment:
According to comment above, duplicate relationships should be removed. Logic below seems to prevent duplicate relationships from being added. Why do we need this? Is the duplicate removal not working?